src/taskchampion-cpp/src/lib.rs

```diff
@@ -0,0 +1,973 @@
+
+        t.delete_task(&mut operations);
+        i += 1;
+        assert_eq!(operations.len(), i + 1);
+        assert!(!operations[i].is_create());
+        assert!(!operations[i].is_update());
+        assert!(operations[i].is_delete());
+        assert!(!operations[i].is_undo_point());
+        assert_eq!(operations[i].get_uuid(), uuid);
+        assert_eq!(
+            operations[i].get_old_task(),
+            vec![ffi::PropValuePair {
+                prop: "prop2".into(),
+                value: "value2".into(),
+            },]
+        );
+    }
+
+    #[test]
+    fn operation_counts() {
+        let mut rep = new_replica_in_memory().unwrap();
+        let mut operations = new_operations();
+        add_undo_point(&mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        add_undo_point(&mut operations);
+        rep.commit_operations(operations).unwrap();
+        // Three non-undo-point operations.
+        assert_eq!(rep.num_local_operations().unwrap(), 3);
+        // Two undo points
+        assert_eq!(rep.num_undo_points().unwrap(), 2);
+    }
+
+    #[test]
+    fn undo_operations() {
+        let mut rep = new_replica_in_memory().unwrap();
+        let mut operations = new_operations();
+        let (uuid1, uuid2, uuid3) = (uuid_v4(), uuid_v4(), uuid_v4());
+        add_undo_point(&mut operations);
+        create_task(uuid1, &mut operations);
+        add_undo_point(&mut operations);
+        create_task(uuid2, &mut operations);
+        create_task(uuid3, &mut operations);
+        rep.commit_operations(operations).unwrap();
+
+        let undo_ops = rep.get_undo_operations().unwrap();
+        assert_eq!(undo_ops.len(), 3);
+        assert!(undo_ops[0].is_undo_point());
+        assert!(undo_ops[1].is_create());
+        assert_eq!(undo_ops[1].get_uuid(), uuid2);
+        assert!(undo_ops[2].is_create());
+        assert_eq!(undo_ops[2].get_uuid(), uuid3);
+    }
+
+    #[test]
+    fn task_lists() {
+        let mut rep = new_replica_in_memory().unwrap();
+        let mut operations = new_operations();
+        add_undo_point(&mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        rep.commit_operations(operations).unwrap();
+
+        assert_eq!(rep.all_task_data().unwrap().len(), 3);
+        assert_eq!(rep.all_task_uuids().unwrap().len(), 3);
+    }
+
+    #[test]
+    fn expire_tasks() {
+        let mut rep = new_replica_in_memory().unwrap();
+        let mut operations = new_operations();
+        add_undo_point(&mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        create_task(uuid_v4(), &mut operations);
+        rep.commit_operations(operations).unwrap();
+        rep.expire_tasks().unwrap();
```

Comment:
I take it we're just testing that expire_tasks doesn't panic?

Reply:
Yes, most of the tests here are basically "doesn't panic" or a kind of quick test for the cxx implementation. The `test/tc.test.cpp` does a better job of actually asserting that things are correct.